make/content/advanced.tex

```diff
@@ -44,16 +45,53 @@
     \end{minted}
   \end{center}
 
-  Lösung: manuell synchronisieren
+  Lösung: Ein \texttt{make}-Feature (\texttt{v4.3}): \emph{grouped targets}
   \begin{center}
-    $\vdots$ \\
+    \small
     \begin{minted}{make}
-      plot1.pdf: plot.py data.txt
+      all: report.txt
+
+      report.txt: plot1.pdf plot2.pdf
+        touch report.txt
+
+      plot2.pdf plot1.pdf &: plot.py data.txt # das &: definiert die targets als group 
           python plot.py
 
-      plot2.pdf: plot1.pdf
     \end{minted}
   \end{center}
+   → Alle \texttt{targets} werden durch eine einzigen Durchlauf der \texttt{recipe} (gebündelt) erzeugt.
+\end{frame}
 
-  Wenn man \texttt{plot2.pdf} aber nicht \texttt{plot1.pdf} löscht, kann \texttt{make} nicht mehr \texttt{plot2.pdf} erstellen.
+\begin{frame}[fragile]{Expert}
+  Wenn ein Skript sehr viele Dateien erzeugt kann die Liste der \texttt{targets} unübersichtlich lang werden.\\
+  Diese \texttt{targets} werden außerdem in der Regel als \texttt{prerequisites} verwendet,
+  z.~B. für die \texttt{recipe} der Berichtdatei  → Die Liste befindet sich sogar zweimal im \texttt{Makefile}
+
+  \begin{center}
+    \small
+    \begin{minted}{make}
+      all: report.txt
+
+      report.txt: plot1.pdf plot2.pdf plot3.pdf plot4.pdf plot5.pdf
+        touch report.txt
+
+      plot1.pdf plot2.pdf plot3.pdf plot4.pdf plot5.pdf: plot.py data.txt
+        python plot.py  # plot.py produziert alle plot{i}.pdf 
+    \end{minted}
+  \end{center}
+  Lösung: Ein weiteres \texttt{make}-Feature: \emph{variables}
+  \begin{center}
+    \small
+    \begin{minted}{make}
+      all: report.txt
+
+      script_targets = plot1.pdf plot2.pdf plot3.pdf plot4.pdf plot5.pdf #Variablen Definition
```

Comment:
Next step: `plots = $(addprefix build/, $(addsuffix .pdf, plot1 plot2 plot3 plot4)`

Reply:
I'm not opposed to add another slide for this to :+1: